src/Common/Uuid/UuidMapping.php

```diff
@@ -29,7 +29,7 @@ public static function createMissingResourceUuids($resource, $table)
             "SELECT `" . $table . "`.`uuid`
                        FROM `" . $table . "`
                        LEFT OUTER JOIN `uuid_mapping` ON `" . $table . "`.`uuid` = `uuid_mapping`.`target_uuid`
-                       WHERE (`uuid_mapping`.`uuid` IS NULL OR `uuid_mapping`.`uuid` = '' OR `uuid_mapping`.`uuid` = '\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0')
+                       WHERE (`uuid_mapping`.`uuid` IS NULL OR `uuid_mapping`.`uuid` = '' OR `uuid_mapping`.`uuid` = '\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0' AND `uuid_mapping`.`resource` != '" . $resource . "')
```

Comment:
should use binding and a parenthesis to help make logic clear (since this is a tough to follow query logic-wise):
```
WHERE ((`uuid_mapping`.`uuid` IS NULL OR `uuid_mapping`.`uuid` = '' OR `uuid_mapping`.`uuid` = '\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0') AND `uuid_mapping`.`resource` != ?)
```

Reply:
Agreed :+1: